helix-core/src/main/java/org/apache/helix/BucketDataAccessor.java

```diff
@@ -0,0 +1,35 @@
+package org.apache.helix;
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.IOException;
+
+
+public interface BucketDataAccessor {
+
+  <T extends HelixProperty> boolean bucketWrite(String path, T value);
+
+  HelixProperty bucketRead(String path);
```

Comment:
`<T extends HelixProperty> T bucketRead(String path, Class<T> detailedPropertyType)` this way help you cast the result to the subclass without IDE warning.

Reply:
There aren't any IDE warnings... but for the sake of consistency (to make it consistent with write), I could also make the read method generic.

Reply:
Sorry I need to clarify more in more detail. What I've noticed in Helix is that the `String path` could point to a specific HelixProperty(IdealState, e.g). Then think about the behavior of the caller, it needs to 
```
IdealState idealState = (IdealState) accessor.bucketRead(IdealStatePath);
```
It'll of course trigger IDE warning (not safe). Casting at the method level will help you remove the warning.